sdk/core/ci.yml

```diff
@@ -42,6 +42,55 @@ stages:
       LiveTestTimeoutInMinutes: 90 # default is 60 min. We need a little longer on worst case for Win+jsonTests
       LineCoverageTarget: 93
       BranchCoverageTarget: 55
+      PreTestSteps:
```

Comment:
Is it just the core pipeline that will need the websocket server? It could make sense eventually to move these tests into their own template and then reference it in `archetype-sdk-*.yml` behind a parameter like:

```
steps:
  - ${{ if parameters.UseWebsocketServer }}:
    - template: /eng/pipelines/templates/steps/start-websocket-server.yml

  ...

  - ${{ if parameters.UseWebsocketServer }}:
    - template: /eng/pipelines/templates/steps/stop-websocket-server.yml
```

Reply:
Today it's just the core pipeline that needs the websocket server. However I could imagine that AMQP is going to need something similar, as well as other possible protocols.

Reply:
@benbp -- We started down a similar route, but since this is scoped just to `core` we're keeping it inline in the `ci.yml` file. 

There will probably be a need for something like what you're proposing in the future for a cross-cutting tool.